scanner/gobin/exe.go

```diff
@@ -0,0 +1,60 @@
+//go:build go1.18
+
+package gobin
+
+import (
+	"context"
+	"debug/buildinfo"
+	"io"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+)
+
+func toPackages(ctx context.Context, out *[]*claircore.Package, p string, r io.ReaderAt) error {
+	bi, err := buildinfo.Read(r)
+	if err != nil {
+		zlog.Info(ctx).
+			Err(err).
+			Msg("unable to open executable")
+		return nil
+	}
+	ctx = zlog.ContextWithValues(ctx, "exe", p)
+	pkgdb := "go:" + p
+
+	*out = append(*out, &claircore.Package{
+		Kind:      claircore.BINARY,
+		Name:      "runtime",
+		Version:   bi.GoVersion,
+		PackageDB: pkgdb,
+	})
+	ev := zlog.Debug(ctx)
+	vs := map[string]string{
+		"runtime": bi.GoVersion,
+	}
+	*out = append(*out, &claircore.Package{
+		Kind:      claircore.BINARY,
+		PackageDB: pkgdb,
+		Name:      bi.Main.Path,
+		Version:   bi.Main.Version,
+	})
+	if ev.Enabled() {
```

Comment:
Is this to save on allocations if debugging isn't enabled?

Reply:
Yes -- no point in building a big map if it's not going to be printed.